libs/components/core/src/lib/modules/affix/dom-utils.ts

```diff
@@ -45,21 +41,72 @@ export function getElementOffset(
   };
 }
 
+/**
+ * Returns an AffixRect that represents the outer dimensions of a given element.
+ */
+export function getOuterRect(element: HTMLElement): Required<AffixRect> {
+  const rect = element.getBoundingClientRect();
+  const computedStyle = window.getComputedStyle(element, undefined);
+  const marginTop = parseInt(computedStyle.marginTop, 10);
```

Comment:
Do you care about decimal values? If so, `parseFloat` might be more appropriate.